src/plugins/intel_cpu/src/snippets_transformations/enforce_precision.cpp

```diff
@@ -0,0 +1,133 @@
+                        break;
+                    }
+                    continue;
+                }
+            }
+            if (!op_is_appropriate) {
+                break;
+            }
+
+            if (op_has_to_be_handled) {
+                supported_precisions_to_enforce = supported_precisions;
+                break;
+            }
+        }
+
+        if (!op_is_appropriate || supported_precisions_to_enforce.empty()) {
+            continue;
+        }
+
+        const auto insert_convert = [](
+            const Output<Node>& parent_output,
+            const std::shared_ptr<Node>& op,
+            const size_t input_index,
+            const element::Type& target) {
+                auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                    parent_output,
+                    target);
+                ngraph::copy_runtime_info(parent_output.get_node_shared_ptr(), convert);
+                op->set_argument(input_index, convert);
+        };
+
+        for (auto index = 0ull; index < supported_precisions_to_enforce.size(); ++index) {
+            if ((supported_precisions_to_enforce[index] == target) || (actual_precisions[index] == source)) {
+                const auto op_parent = ov::as_type_ptr<ngraph::snippets::op::ConvertSaturation>(op->get_input_node_shared_ptr(index));
+                if ((op_parent != nullptr) &&
+                    (op_parent->get_input_element_type(0) == target) &&
+                    // we can remove existing convertion only if precisions before and after are appropriate for removal
+                    snippets::pass::PropagatePrecision::can_be_removed(
+                        op_parent->get_input_element_type(0),
+                        actual_precisions[index],
+                        supported_precisions_to_enforce[index])) {
+                    // remove convert
+                    op_parent->output(0).replace(op_parent->get_input_source_output(0));
+                    was_updated = true;
+                } else if (supported_precisions_to_enforce[index] != actual_precisions[index]) {
+                    insert_convert(op->get_input_source_output(index), op, index, target);
+                    was_updated = true;
+                }
+            }
+        }
+
+        auto type_relaxed_node = std::dynamic_pointer_cast<ov::op::TypeRelaxedBase>(op);
+        if (was_updated || (type_relaxed_node != nullptr)) {
+            const bool res = ngraph::snippets::pass::PropagatePrecision::validate_and_infer_types_and_restore_outputs(op);
+            was_updated = was_updated || res;
+        }
+    }
+
+    return was_updated;
+}
+
+EnforcePrecision::Operation EnforcePrecision::get_supported_precisions_default(
+    const std::shared_ptr<ngraph::Node>&op) noexcept {
+    if (ov::is_type<ngraph::snippets::op::Brgemm>(op)) {
+        return EnforcePrecision::Operation{ true, {{element::bf16, element::bf16}} };
+    }
```

Comment:
I suggest to remove `Operation::is_target_isa_supported` and make so if you want to have generic solution with common interface without additional class fields:
```suggestion
    const std::shared_ptr<ngraph::Node>&op) noexcept {
    if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
        if (ov::is_type<ngraph::snippets::op::Brgemm>(op)) {
            return EnforcePrecision::Operation{{{element::bf16, element::bf16}} };
        }
    }
```
Because the pass is plugin specific.

What do you think?

Reply:
customization removal doesn't make a solution more generic. Was simplified.

Reply:
done thanks